docs/development.md

```diff
@@ -0,0 +1,17 @@
+# Development
+
+## Compile flags
+
+In order to fix Intellisense issues in VSCode and several other IDEs related to missing header files in C++ dependencies such as `ada-url` and `simdutf`, you need to generate a `compile_commands.json` document. Follow the steps below to fix the issue:
```

Comment:
Strictly speaking this is not true: compile_flags.txt is enough for clangd to resolve includes. What doesn't work is find references, since clangd can't enumerate all files in the project.
You need to generate compile_commands.json only if you want these additional features to work.

Reply:
Would you mind recommending a replacement for this line?

Reply:
```
## Clangd

To get language support in VSCode and other IDEs we recommend using `clangd`-based language server.
The server reads supplied `compile_flags.txt` file for correct options and include paths to resolve symbols
in opened files.

To support `clangd` project-level operations like `Find References` `compile_commands.json` file needs to
be generated that will list all files in the project. If you do this, then it needs to be re-generated periodically
when new files appear.

....
```